File: 2025/01/05/stringify.cpp
#include <iostream>
#include <string>
#include "json.hpp"

int main() {
  // 定义变量
  std::string message = "Success";
  std::string uID = "1000001";
  std::string username = "李燃";
  std::string authority_num = "";

  // 构造 JSON 对象
  nlohmann::json output = {
    {"message", message},
    {"uID", uID},
    {"username", username},
    {"authority_num", authority_num}
  };

  // 输出 JSON 对象
  std::cout << output << std::endl;
  return 0;
}

File: 2025/01/05/stringify.js
const { execFile } = require('child_process');
const path = require('path');

const mainPath = './main';
execFile(mainPath, (error, stdout, stderr) => {
  if (error) {
    console.error(`执行错误: ${error.message}`);
  }
  if (stderr) {
    console.error(`标准错误: ${stderr}`);
  }
  const mainOutput = JSON.parse(stdout);
  console.log(mainOutput.uID);
  console.log(mainOutput.message);
  console.log(mainOutput.username);
  console.log(mainOutput.authority_num);
});
